crates/router/src/connector/cybersource/transformers.rs

```diff
@@ -871,134 +871,173 @@ impl TryFrom<&CybersourceRouterData<&types::PaymentsAuthorizeRouterData>>
+                                        )
+                                    });
+
+                                Ok(Self {
+                                    processing_information,
+                                    payment_information,
+                                    order_information,
+                                    client_reference_information,
+                                    merchant_defined_information,
+                                    consumer_authentication_information: None,
+                                })
+                            }
                         }
                     }
+                    payments::WalletData::GooglePay(google_pay_data) => {
+                        Self::try_from((item, google_pay_data))
+                    }
+                    payments::WalletData::AliPayQr(_)
+                    | payments::WalletData::AliPayRedirect(_)
+                    | payments::WalletData::AliPayHkRedirect(_)
+                    | payments::WalletData::MomoRedirect(_)
+                    | payments::WalletData::KakaoPayRedirect(_)
+                    | payments::WalletData::GoPayRedirect(_)
+                    | payments::WalletData::GcashRedirect(_)
+                    | payments::WalletData::ApplePayRedirect(_)
+                    | payments::WalletData::ApplePayThirdPartySdk(_)
+                    | payments::WalletData::DanaRedirect {}
+                    | payments::WalletData::GooglePayRedirect(_)
+                    | payments::WalletData::GooglePayThirdPartySdk(_)
+                    | payments::WalletData::MbWayRedirect(_)
+                    | payments::WalletData::MobilePayRedirect(_)
+                    | payments::WalletData::PaypalRedirect(_)
+                    | payments::WalletData::PaypalSdk(_)
+                    | payments::WalletData::SamsungPay(_)
+                    | payments::WalletData::TwintRedirect {}
+                    | payments::WalletData::VippsRedirect {}
+                    | payments::WalletData::TouchNGoRedirect(_)
+                    | payments::WalletData::WeChatPayRedirect(_)
+                    | payments::WalletData::WeChatPayQr(_)
+                    | payments::WalletData::CashappQr(_)
+                    | payments::WalletData::SwishQr(_) => {
+                        Err(errors::ConnectorError::NotImplemented(
+                            utils::get_unimplemented_payment_method_error_message("Cybersource"),
+                        )
+                        .into())
+                    }
+                },
+                // If connector_mandate_id is present MandatePayment will be the PMD, the case will be handled in the first `if` clause.
+                // This is a fallback implementation in the event of catastrophe.
+                payments::PaymentMethodData::MandatePayment => {
+                    let processing_information = ProcessingInformation::from((item, None));
```

Comment:
I understand why we need this block, but lets have it inside try_from so that we can avoid code repetition